packages/esm-appointments-app/src/hooks/useClinicalMetrics.ts

```diff
@@ -45,11 +45,13 @@ export function useAllAppointmentsByDate() {
     openmrsFetch,
   );
 
-  const providersArray = data?.data?.filter(({ providers }) => providers !== null) ?? [];
-  const providersCount = uniqBy(
-    providersArray.map(({ providers }) => providers).flat(),
-    (provider) => provider.uuid,
-  ).length;
+  const providersArray = data?.data?.flatMap(({ providers }) => providers ?? []) ?? [];
+
+  const validProviders = providersArray.filter((provider) => provider.response === 'ACCEPTED');
+
+  const uniqueProviders = uniqBy(validProviders, (provider) => provider.uuid);
```

Comment:
Thanks @lucyjemutai , whats the use case for this?

Reply:
@CynthiaKamau  When an appointment is edited and the provider is changed, the new provider's response is updated to "ACCEPTED," while the previously assigned provider's status is changed to "CANCELLED." All providers ever associated with that patient are listed, including those with outdated statuses. So, to ensure we only count the active providers for the current day, have filter out those with a status other than 'ACCEPTED'. This way, we accurately display the count of active providers per day. For example, in the payload preview-->
_"name": "John Khmer",
  "providers": [
    { "uuid": "8b97c410-67c5-4e49-9a93-265a23b4dd45", "response": "CANCELLED", "name": "Kelvin" },
    { "uuid": "705f5791-07a7-44b8-932f-a81f3526fc98", "response": "ACCEPTED", "name": "Jane Nurse" },
    { "uuid": "f39e57d8-1185-4199-8567-6f1eeb160f05", "response": "CANCELLED", "name": "Super User" },
    { "uuid": "37b8d5ab-49a5-460d-947d-1a385dfb56f2", "response": "CANCELLED", "name": "Bad Man Josh" }
  ]_

Reply:
Thanks for the clarification